daxa/mission/erosita.py

```diff
@@ -334,72 +296,75 @@ def _inst_filtering(self, evlist_path: str):
         """
         Method to filter event lists for eROSITACalPV data based on instrument choice.
         
-        :param Union[List[str], str] gd_tscopes: The names of telescopes from which the user wishes to INCLUDE data from.
+        :param str evlist_path: This is the file path to the raw eventlist for a certain ObsID
+         that has NOT been filtered for the users instrument choice yet.
         """
 
         insts = self.chosen_instruments
-        # Getting a string of the numbers of the telescope modules to add to the file name
+
+        # Getting a string of TM numbers to add to the end of the file name
         insts_str = ''.join(sorted(re.findall(r'\d+', ''.join(insts))))
 
-        # Only filtering the eventlist if this combination of instruments hasn't been filtered for yet
+        # Checking that this combination of instruments has not been filtered for before for this Obsid
+        #  this is done by checking that there is no file with the _if_{}.fits ending where {} is the 
+        #  number(s) of the TM(s) that the user has specified when declaring the mission.
+        # Indexing the string at [:-5] removes the .fits part of the file path
         if os.path.exists(evlist_path[:-5] + '_if_{}.fits'.format(insts_str)):
             pass
         else:
```

Comment:
I noticed the first time I reviewed the pull request (but forgot to mention) that if data are already downloaded (cached I guess we would call it), and then another run of eROSITACalPV is done but with instruments filtered, the requested event lists aren't created.

Reply:
Ahh I see what you mean, it will only filter instruments when the download() is called at the moment, I will change how _inst_filtering() works, and then call it earlier

Reply:
ok i had a thought, 

since I parallelised the inst_filtering function, shouldnt it be a static method like _download_call?

Then if I want to do that, I will leave the inst_filtering function as it was (except now static), and instead overwrite the chosen_instruments function

Reply:
Ok cool, I've committed this change, if you see this before monday I haven't tested it yet, so be careful!